internal/worker/uniter/runner/context/context.go

```diff
@@ -1405,12 +1409,19 @@ func (ctx *HookContext) HookVars(
 		return nil, errors.Trace(err)
 	}
 
+	if scope := coretrace.SpanFromContext(ctx).Scope(); scope.TraceID() != "" {
+		vars = append(vars,
+			"JUJU_TRACE_ID="+scope.TraceID(),
+			"JUJU_SPAN_ID="+scope.SpanID(),
+		)
+	}
```

Comment:
The actual change.

Note: we should pass in if the trace is being recorded, but I need to change some code around to support that.